### ISSUE TITLE
Make `msgpack.encode` return `varbinary` object

### ISSUE THREAD

**Issue by locker** [tarantool/tarantool#10351]
MsgPack data is a binary string so `msgpack.encode` should return a `varbinary` object, not a plain string. Since this may break application code, we should introduce a new `compat` module option for this change.

**Comment by locker** [comment]
Blocked by #10350.

**Comment by locker** [comment]
This change breaks 19 our internal tests because they expect that an object returned by a `msgpack.encode` has string methods (like `string.sub`). I see the following options:
- Simply fix the tests. Still, we may experience a lot of similar breakages from our users so this solution is dubious.
- Make `varbinary` behave like a string, i.e. add the corresponding methods (at least those that make sense for a binary string) and make our internal functions that take binary strings (like socket write) accept `varbinary` objects as well.
- Leave everything as is and close this issue as "Won't fix". That is, our internal functions will continue to work only with Lua strings while `varbinary` will be used exclusively for encoding to `MP_BIN`.

**Comment by locker** [comment]
Reworking all our internal functions to work with `varbinary` would be an overkill while other solutions are ugly so I guess I'll just drop this one.
